apps/site/pages/en/learn/modules/how-to-use-streams.md

```diff
@@ -0,0 +1,599 @@
+- **Flow Control**: They inherently manage backpressure, as the consumer controls the flow by awaiting the next piece of data, allowing for more efficient memory usage and processing.
+
+Async iterators offer a more modern and often more readable way to work with readable streams, especially when dealing with asynchronous data sources or when you prefer a more sequential, loop-based approach to data processing.
+
+Here's an example demonstrating the use of async iterators with a readable stream:
+
+```mjs
+import fs from 'fs';
+import { pipeline } from 'stream/promises';
+
+await pipeline(
+  fs.createReadStream(import.meta.filename),
+  async function* (source) {
+    for await (let chunk of source) {
+      yield chunk.toString().toUpperCase();
+    }
+  },
+  process.stdout
+);
+```
+
+This code achieves the same as the previous examples, without the need to define a new transform stream. The error from previous examples has been removed for the sake of brevity. The async version of pipeline has been used. It should be wrapped in a `try..catch` block to handle possible errors.
+
+### Object mode
+
+By default streams can work with strings, [`Buffer`], [`TypedArray`], or [`DataView`], if an arbitrary value different than those (e.g. an object) is pushed into a stream a `TypeError` will be thrown. However, it is possible to work with objects by setting the `objectMode` option to `true`. This allows the stream to work with any JavaScript value, with the exception of `null`, which is used to signal the end of the stream. This means you can `push` and `read` any value in a readable stream and `write` any value in a writable stream.
+
+```cjs
+const { Readable } = require('stream');
+
+const readable = Readable({
+  objectMode: true,
+  read() {
+    this.push({ hello: 'world' });
+    this.push(null);
+  },
+});
+```
+
+When working in object mode it is important to remember that the `highWaterMark` option is the number of objects, not bytes.
+
+### Backpressure
+
+When using streams it is important to make sure the producer doesn't overwhelm the consumer. For this the backpressure mechanism is used in all streams in the Node.js API and implementors are responsible to keep that behaviour.
+
+In any scenario where the data buffer has exceeded the [`highWaterMark`](https://nodejs.org/api/stream.html#stream_buffering) or the write queue is currently busy, [`.write()`](https://nodejs.org/api/stream.html#stream_writable_write_chunk_encoding_callback) will return `false`.
+
+When a `false` value is returned, the backpressure system kicks in. It will pause the incoming [`Readable`](https://nodejs.org/api/stream.html#stream_readable_streams) stream from sending any data and wait until the consumer is ready again. Once the data buffer is emptied, a [`'drain'`](https://nodejs.org/api/stream.html#stream_event_drain) event will be emitted and resume the incoming data flow.
+
+For a deeper understanding of backpressure, check the [`backpressure guide`][].
```

Comment:
```suggestion
When using streams, it is important to make sure the producer doesn't overwhelm the consumer. For this, the backpressure mechanism is used in all streams in the Node.js API, and implementors are responsible for maintaining that behavior.

In any scenario where the data buffer has exceeded the [`highWaterMark`](https://nodejs.org/api/stream.html#stream_buffering) or the write queue is currently busy, [`write()`](https://nodejs.org/api/stream.html#stream_writable_write_chunk_encoding_callback) will return `false`.

When a `false` value is returned, the backpressure system kicks in. It will pause the incoming [`Readable`](https://nodejs.org/api/stream.html#stream_readable_streams) stream from sending any data and wait until the consumer is ready again. Once the data buffer is emptied, a [`'drain'`](https://nodejs.org/api/stream.html#stream_event_drain) event will be emitted to resume the incoming data flow.

For a deeper understanding of backpressure, check the [`backpressure guide`][].
```